src/d/parser/expression.d

```diff
@@ -1047,3 +1050,78 @@ StringLiteral parseStringLiteral(ref TokenRange trange) {
 	
 	return new StringLiteral(location, name.toString(trange.context));
 }
+
+/**
+ * Parse floating point literals
+ */
+FloatLiteral parseFloatLiteral(ref TokenRange trange) {
+	const location = trange.front.location;
+	auto litString = trange.front.toString(trange.context);
+
+	trange.match(TokenType.FloatLiteral);
+	import d.common.builtintype : BuiltinType;
+
+	//Look for a suffix
+	if (litString.length > 1)
+	{
+		switch (litString[$-1])
+		{
+			// https://dlang.org/spec/lex.html#FloatSuffix
+			import std.conv : to;
+			case 'f':
+			case 'F':
+				const float f = litString[0..$-1].to!float;
+				return new FloatLiteral(location, f, BuiltinType.Float);
+			case 'L':
+				import source.exception;
+				throw new CompileException(location, "SDC does not support real literals yet");
+			default:
+				//Lexed correctly but no suffix, it's a double
+				const double d = litString[0..$].to!float;
+				return new FloatLiteral(location, d, BuiltinType.Double);
+		}
+	}
+	assert(0);
+}
+
+@("Test FloatLiteral parsing")
+unittest
+{
+	import util.testinfra;
+	import source.context;
+	auto context = new Context();
+	import source.parserutil;
+	auto tokensFromString(string s, bool skipBegin = true) {
+		import source.name;
+		auto base = context.registerMixin(Location.init, s ~ '\0');
+		auto x = lex(base, context);
+		if (skipBegin)
+		{
+			x.match(TokenType.Begin);
+		}
+		return x;
+	}
+	void floatRoundTrip(T)(const string floatString, const T floatValue)
+		if (__traits(isFloating, T))
+	{
+		import d.common.builtintype : BuiltinType;
+		const BuiltinType expectedType = is(T == float) ? BuiltinType.Float : BuiltinType.Double;
+		auto tr = tokensFromString(floatString);
+		const result = parseExpression(tr);
```

Comment:
Just calling parseFloatLiteral is fine here, IMO

If there is a problem with that in parseExpression, we are goign to know real quick and all kind of other tests are going to break, so I think this is fine.

Reply:
To clarify, the point of test is not just to know what works and what doesn't, but also help pin point exactly what breaks. You don't really want to see this specific test break when parseExpression is broken if parseFloatLiteral  works fine.

Reply:
Good point.